src/exchanges.js

```diff
@@ -124,20 +116,20 @@ var Publisher = function(conn, channel, entries, exchangePrefix, options) {
           contentEncoding:    'utf-8',
           CC:                 CCs
         }, function(err, val) {
-          if (that._reporter) {
+          if (monitor) {
             // Get duration
             var d = process.hrtime(start);
 
-            // Report statistics
-            that._reporter({
-              component:      options.component,
-              process:        options.process,
-              duration:       d[0] * 1000 + (d[1] / 1000000),
-              routingKeys:    1 + CCs.length,
-              payloadSize:    payload.length,
-              exchange:       exchange,
-              error:          (err ? 'true' : 'false'),
-            });
+            var success = 'success';
+            if (err) {
+              success = 'error';
+            }
+
+            for (let stat of [success, 'all']) {
+              var k = [exchange, stat].join('.');
```

Comment:
I'm tempted to suggest we use `entry.name` instead... `entry.name` is the identifier for the exchange... It's shorter and you can lookup the exact exchange in docs...

Note: I think `'all'` is too much.. we'll rarely have errors here... It's nice to have timing and count of errors, but I doubt we need timing for "error" and "success" combined... that's probably overkill...